### PR TITLE
Implemented HW5

### PR DESCRIPTION
**Pull request by stekkko**
Don`t know how to do exercise 5 or 6. Pls help :-)

homework 5:
https://www.seas.upenn.edu/~cis194/spring13/hw/05-type-classes.pdf

### CODE REVIEW

File: src/Calc.hs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+testSat     = testExp :: Maybe Mod7
+
+--exersize 6
+class HasVars a where
+  var :: String -> a
+
+data VarExprT = Var String Integer 
+           | Lit Integer
+           | Add ExprT ExprT
+           | Mul ExprT ExprT
```
Comment (chshersh) [inline src/Calc.hs]: In your module `ExprT` you already have constructors with names like `Add`, `Lit`, `Mul`, so this won't work.

**Review by chshersh — COMMENTED** [review]
Currently looks good 👍

File: src/Calc.hs (hunk 1 above)
Comment (stekkko) [reply src/Calc.hs]: What should I do?

Comment (chshersh) [reply src/Calc.hs]: @MINER9000 You can rename constructors. That should work :+1:

**Review by chshersh — APPROVED** [review]
Great!

**Pull request merged**